packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -407,8 +408,9 @@ function buildBodyParameter(
           bodyParameter.clientName
         }, "${getEncodingFormat(bodyParameter.type)}")`;
   } else if (
-    bodyParameter.type.type === "byte-array" &&
-    bodyParameter.type.format === "binary"
+    (bodyParameter.type.type === "byte-array" &&
+      bodyParameter.type.format === "binary") ||
+    bodyParameter.isBinaryPayload
```

Comment:
It seems that when `bodyParameter.type.type === "byte-array" &&
      bodyParameter.type.format === "binary"` is also binary, why not enhance the isBinaryPayload logic to cover that case?

Reply:
Changed both to use isBinaryPayload and fixed a small bug in isByteType.